bitly/src/bitly_short_link_handler.erl

```diff
@@ -0,0 +1,30 @@
+-module(bitly_short_link_handler).
+
+-export([init/2]).
+
+init(Req, State) ->
+    Url = get_request_url(Req),
+    case cowboy_req:method(Req) of
+        <<"POST">> ->
+            {Status, ShortUrl} = handle_post(Url);
```

Comment:
Just an idea for your consideration; you can do something like:

```
Method = cowboy_req:method(Req),
{Status, ShortUrl} = handle(Method, Url),
```

This way you use different `handle` clauses for each method, which I think is more elegant. (I think there's a rule of thumb which is that frequently it's a good idea to extract an inner case into a helper function with different clauses)

Reply:
Good catch